core-plugins/src/main/java/co/cask/hydrator/plugin/common/ExcelInputFormat.java

```diff
@@ -0,0 +1,187 @@
+import org.apache.poi.hssf.usermodel.HSSFWorkbook;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.Row;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Iterator;
+
+
+/**
+ * {@link ExcelInputFormat} is {@link TextInputFormat} implementation for reading Excel files.
+ *
+ * The {@link ExcelInputFormat.ExcelRecordReader} reads a given sheet, and within a sheet reads
+ * all columns and all rows.
+ */
+public class ExcelInputFormat extends TextInputFormat {
+  private static final Logger LOG = LoggerFactory.getLogger(ExcelInputFormat.class);
+
+  @Override
+  public RecordReader<LongWritable, Text> createRecordReader(InputSplit split, TaskAttemptContext context) {
+    return new ExcelRecordReader();
+  }
+
+  /**
+   * Reads excel spread sheet, where the keys are the offset in the excel file and the text is the complete record.
+   */
+  public static class ExcelRecordReader extends RecordReader<LongWritable, Text> {
+    // Map key that represents the row index.
+    private LongWritable key;
+
+    // Map value that represents : // {file}\r{sheet}\r{type}\t{col}\t{value}\r{type}\r{col}\r{value}...
+    private Text value;
+
+    // Represents a indvidual cell
+    private Cell cell;
+
+    // Specifies all the rows of an Excel spreadsheet - An iterator over all the rows.
+    private Iterator<Row> rows;
+
+    // InputStream handler for Excel files.
+    private FSDataInputStream fileIn;
+
+    // Path of input file.
+    private Path file;
+
+    // Specifies the row index.
+    private long rowIdx;
+
+    // Specifies last row num.
+    private int lastRowNum;
+
+    @Override
+    public void initialize(InputSplit genericSplit, TaskAttemptContext context) throws IOException,
+      InterruptedException {
+
+      FileSplit split = (FileSplit) genericSplit;
+      Configuration job = context.getConfiguration();
+      file = split.getPath();
+
+      FileSystem fs = file.getFileSystem(job);
+      fileIn = fs.open(split.getPath());
+
+      // Reads the excel file, selects the sheet to be read.
+      HSSFWorkbook workbook = new HSSFWorkbook(fileIn);
+
+      String sheetName = job.get("excel.sheet.name");
+      int sheetIndx = job.getInt("excel.sheet.index", -1); // Index overrides name.
+
+      HSSFSheet sheet;
+      if (sheetIndx < 0 && sheetName != null && !sheetName.isEmpty()) {
+        sheet = workbook.getSheet(sheetName);
+      } else {
+        sheet = workbook.getSheetAt(sheetIndx);
+      }
+
+      rows = sheet.rowIterator();
+      lastRowNum = sheet.getLastRowNum();
+      rowIdx = 0;
+
+      boolean skipFirstRow = job.getBoolean("excel.sheet.skip.first.row", false);
+      if (skipFirstRow && rows.hasNext()) {
+        rows.next(); // Reads the first row.
+        rowIdx = 1;
+      }
+    }
+
+    @Override
+    public boolean nextKeyValue() throws IOException, InterruptedException {
+      if(! rows.hasNext()) {
```

Comment:
missing whitespace after if, should not have whitespace after !.

Is your checkstyle configured properly?